pkg/signature/ecdsa.go

```diff
@@ -190,8 +192,21 @@ func (e ECDSAVerifier) VerifySignature(signature, message io.Reader, opts ...Ver
 		return fmt.Errorf("invalid ECDSA public key for %s", e.publicKey.Params().Name)
 	}
 
-	if !ecdsa.VerifyASN1(e.publicKey, digest, sigBytes) {
-		return errors.New("invalid signature when validating ASN.1 encoded signature")
+	var asnParseTest []*big.Int
+	if _, err := asn1.Unmarshal(sigBytes, &asnParseTest); err == nil {
```

Comment:
Is this the correct check? I would have thought that a sequence needs to be represented as a struct?

Reply:
Maybe it's just flexible enough since it is effectively an array. Seems a little brittle, but if the tests pass, this should be sufficient.

Reply:
This is what was recommended by https://pkg.go.dev/encoding/asn1#Unmarshal